to txt/config.txt
# config.py
import re

# ----- Имя базы -----
BASE_NAME = 'Cable base ver.'

# ----- Регулярные выражения -----

regular_journal_kks = re.compile(r'AKU\.\d{2,4}\.\d{2}[A-Z]{3}\.[A-Z0-9.]+-[A-Z0-9]+')     # ККС журнала
regular_journal_kks_short = re.compile(r'AKU\.\d{2,4}\.\d{2}[A-Z]{3}\.[A-Z0-9.]+(?:-[A-Z0-9]+)?') # ККС журнала с возможным отсутствием хвоста -EMB0001

# ----- Регулярное выражение для имени файла журнала -----
regular_filename = re.compile(
    r'^(?P<kks>AKU\.\d{2,4}\.\d{2}[A-Z]{3}\.[A-Z0-9.]+-[A-Z0-9]+)'  # ККС журнала
    r'(?:_(?P<source>ВКРД|ВКИИ|СРД|СИИ))?'                          # шифр источника (опционально)
    r'(?:_(?P<date>\d{2}\.\d{2}\.\d{4}))?'                          # дата (опционально)
    r'(?:\.docx?)?$'                                                # расширение файла
)

regular_num = re.compile(r"\d+(?:[.,]\d+)?") # re.compile("\\d+[.,]?\\d+")                      # Номер

number_section = r'\d+(?:[,.]\d+)?'
regular_cableSection = re.compile(rf"\d?[(]?{number_section}[xх×]{number_section}[)]?(?:[xх×]{number_section})*")  # Сечение
regular_cableMark = re.compile("[н][г][' ']?[(/]?[AАС]?[)/]?[--]")                  # Марка кабеля 2


regular_KKS = re.compile("\\d{2,3}[А-ЯA-Z]{2,3}\\d{2,3}")             # ККС оборудования 10BAT01GH301
regular_KKS_any = re.compile(r'(?=.*[A-Z])(?=.*\d)[A-Z\d\s.-]{5,}')   # ККС оборудования или помещения

# Часть строки
regular_KKS_room = re.compile(r'\d{2}[A-Z]{3}\d{1,3}R\d{1,3}')  # ККС помещения
regular_KKS_building = re.compile(r'\d{2}[A-Z]{3}')             # ККС здания
regular_KKS_equipment = re.compile(r'(?=.*[A-Z])(?=.*\d)(?!.*-.*-)(?!.*\s.*\s)[A-Z\d\s.-]{5,}') # ККС оборудования
regular_cable_tray = re.compile(r'\d{2}\.\d{2}[A-Z]{3}\d{2}\.\d{2,3}(?:/\d{1,3})?') # ККС конструкции 00.07UBG10.105/4 или 00.07UBG10.105 

# Вся строка
regular_KKS_room_full = re.compile(r'^\d{2}[A-Z]{3}\d{1,3}R\d{1,3}$')  # ККС помещения
regular_KKS_building_full = re.compile(r'^\d{2}[A-Z]{3}$')  # ККС здания
regular_KKS_equipment_full = re.compile(r'^(?=.*[A-Z])(?=.*\d)(?!.*-.*-)(?!.*\s.*\s)[A-Z\d\s.-]{5,}$') # ККС оборудования

regular_axis_full = re.compile(r'^[+-]?\d{1,8}\.?,?\d{0,4}\.?$')   # координата целиком
regular_axis = re.compile(r'[+-]?\d{1,8}\.?,?\d{0,4}\.?')   # координата как часть текста

regular_letter_minus = re.compile(r'[a-zA-Zа-яА-ЯёЁ-]')  # хотя бы одна буква или знак -

# ----- Регулярное выражение для имени журнала -----
common_start = r'([А-ЯA-Z]{3})\.([0-9]{4})\.([0-9]{2})'
type1 = r'([А-ЯA-Z]{3})\.([А-ЯA-Z]{3})\.([А-ЯA-Z]{2})\.([А-ЯA-Z]{2})'        # UFC.CYY.SS.MB
type2 = r'([А-ЯA-Z]{3})\.([0-9]{1})\.([А-ЯA-Z]{2})\.([А-ЯA-Z]{2})'            # UKS.0.AP.MB
type3 = r'([0-9]{1})\.([А-ЯA-Z]{3})\.([А-ЯA-Z]{2})\.([А-ЯA-Z]{2})'            # 0.UKS.AP.MB
number_part = r'([0-9]{4})'
suffix = rf'[-_]([А-ЯA-Z]{{3}})([0-9]{{4}})'  # необязательный суффикс
regular_Journal = re.compile(rf'{common_start}(?:{type1}|{type2}|{type3}){number_part}(?:{suffix})?')










arrayCableMarks = ['КУГПЭПнг', 
                   'SM-MLT', 
                   'КППГЭнг', 
                   'эп(А)-', 
                   'нг(А)', 
                   'КГН', 
                   'КРА6.560.106-27', 
                   'EC4-50-HF', 
                   'TППэп(A)-HДГ',
                   'СПЕЦЛАН',
                   'Complete',
                   'UNITRONIC',
                   'ТАДУ',
                   'KCнг(A)-HF',
                   '6XV1871-2F',
                   'Hyperline',
                   'СКАБ',
                   'PK75',
                   'PROFINET',
                   'F/UTP',
                   'FO-ST-IN/OUT',
                   'КАГЭнг(B)',
                   'нг(A)-HF',
                   'LLMD',
                   'ТК-1, тип Б исп.45',
                   'ТК-1, тип Б',
                   'Спецкабель',
                   'ТППэп',
                   'нг(A)',
                   'КПЭТИнг(B)-HF-Т',
                   'РКГМ',
                   'ППГнг(A)-HF',
                   'ОКБнг(В)-',
                   'OKБнг(B)-',
                   'ОБР-У-4ОВ SM',
                   'КУГПвЭПвнг(A)-HF-Т',
                   'КПЭТИнг(B)-FRHF-Т',
                   'RG58',
                   'RG213',
                   'KPEPng(А)-FRHF-0.25',
                   'KVVGEng(A)-FRLS',
                   'КПСВЭВнг(А)-LS'
                   ]


# ----- Заголовки таблицы Excel -----
HEADERS = [
    [0, 'Журнал', 40],
    [1, 'Номер кабеля', 9],
    [2, 'ККС', 21],
    [3, 'Группа', 7],
    [4, 'Марка', 22],
    [5, 'Сечение', 10],
    [6, 'Диаметр', 8],
    [7, 'Длина', 6],
    [8, 'Трасса', 20],
    [9, 'Откуда помещение', 14],
    [10, 'Откуда оборудование', 20],
    [11, 'Откуда x', 11],
    [12, 'Откуда y', 11],
    [13, 'Откуда z', 11],
    [14, 'Куда помещение', 14],
    [15, 'Куда оборудование', 20],
    [16, 'Куда x', 11],
    [17, 'Куда y', 11],
    [18, 'Куда z', 11],
    [19, 'Резервирование', 8],
    [20, 'Дата добавления в базу', 10],
    [21, 'Версия базы', 7],
    [22, 'id кабеля', 8],
    [23, 'Номер ревизии', 6],
    [24, 'Дата ревизии', 10],
    [25, 'Источник информации', 8],
    [26, 'Класс безопасности', 7],
    [27, 'Статус прокладки', 10],
    [28, 'Статус объединения', 10],
    [29, 'Raw', 25],
    [30, 'min длина', 10],
    [31, 'Требования к объединению', 20],
    [32, 'Ответная часть (из КЖ)', 40],
    [33, 'Наличие в базе', 20],    
    [34, 'Примечание', 40]
]

config.py
# config.py
import re

# ----- Имя базы -----
BASE_NAME = 'Cable base ver.'

# ----- Регулярные выражения -----

regular_journal_kks = re.compile(r'AKU\.\d{2,4}\.\d{2}[A-Z]{3}\.[A-Z0-9.]+-[A-Z0-9]+')     # ККС журнала
regular_journal_kks_short = re.compile(r'AKU\.\d{2,4}\.\d{2}[A-Z]{3}\.[A-Z0-9.]+(?:-[A-Z0-9]+)?') # ККС журнала с возможным отсутствием хвоста -EMB0001

# ----- Регулярное выражение для имени файла журнала -----
regular_filename = re.compile(
    r'^(?P<kks>AKU\.\d{2,4}\.\d{2}[A-Z]{3}\.[A-Z0-9.]+-[A-Z0-9]+)'  # ККС журнала
    r'(?:_(?P<source>ВКРД|ВКИИ|СРД|СИИ))?'                          # шифр источника (опционально)
    r'(?:_(?P<date>\d{2}\.\d{2}\.\d{4}))?'                          # дата (опционально)
    r'(?:\.docx?)?$'                                                # расширение файла
)

regular_num = re.compile(r"\d+(?:[.,]\d+)?") # re.compile("\\d+[.,]?\\d+")                      # Номер

number_section = r'\d+(?:[,.]\d+)?'
regular_cableSection = re.compile(rf"\d?[(]?{number_section}[xх×]{number_section}[)]?(?:[xх×]{number_section})*")  # Сечение
regular_cableMark = re.compile("[н][г][' ']?[(/]?[AАС]?[)/]?[--]")                  # Марка кабеля 2


regular_KKS = re.compile("\\d{2,3}[А-ЯA-Z]{2,3}\\d{2,3}")             # ККС оборудования 10BAT01GH301
regular_KKS_any = re.compile(r'(?=.*[A-Z])(?=.*\d)[A-Z\d\s.-]{5,}')   # ККС оборудования или помещения

# Часть строки
regular_KKS_room = re.compile(r'\d{2}[A-Z]{3}\d{1,3}R\d{1,3}')  # ККС помещения
regular_KKS_building = re.compile(r'\d{2}[A-Z]{3}')             # ККС здания
regular_KKS_equipment = re.compile(r'(?=.*[A-Z])(?=.*\d)(?!.*-.*-)(?!.*\s.*\s)[A-Z\d\s.-]{5,}') # ККС оборудования
regular_cable_tray = re.compile(r'\d{2}\.\d{2}[A-Z]{3}\d{2}\.\d{2,3}(?:/\d{1,3})?') # ККС конструкции 00.07UBG10.105/4 или 00.07UBG10.105 

# Вся строка
regular_KKS_room_full = re.compile(r'^\d{2}[A-Z]{3}\d{1,3}R\d{1,3}$')  # ККС помещения
regular_KKS_building_full = re.compile(r'^\d{2}[A-Z]{3}$')  # ККС здания
regular_KKS_equipment_full = re.compile(r'^(?=.*[A-Z])(?=.*\d)(?!.*-.*-)(?!.*\s.*\s)[A-Z\d\s.-]{5,}$') # ККС оборудования

regular_axis_full = re.compile(r'^[+-]?\d{1,8}\.?,?\d{0,4}\.?$')   # координата целиком
regular_axis = re.compile(r'[+-]?\d{1,8}\.?,?\d{0,4}\.?')   # координата как часть текста

regular_letter_minus = re.compile(r'[a-zA-Zа-яА-ЯёЁ-]')  # хотя бы одна буква или знак -

# ----- Регулярное выражение для имени журнала -----
common_start = r'([А-ЯA-Z]{3})\.([0-9]{4})\.([0-9]{2})'
type1 = r'([А-ЯA-Z]{3})\.([А-ЯA-Z]{3})\.([А-ЯA-Z]{2})\.([А-ЯA-Z]{2})'        # UFC.CYY.SS.MB
type2 = r'([А-ЯA-Z]{3})\.([0-9]{1})\.([А-ЯA-Z]{2})\.([А-ЯA-Z]{2})'            # UKS.0.AP.MB
type3 = r'([0-9]{1})\.([А-ЯA-Z]{3})\.([А-ЯA-Z]{2})\.([А-ЯA-Z]{2})'            # 0.UKS.AP.MB
number_part = r'([0-9]{4})'
suffix = rf'[-_]([А-ЯA-Z]{{3}})([0-9]{{4}})'  # необязательный суффикс
regular_Journal = re.compile(rf'{common_start}(?:{type1}|{type2}|{type3}){number_part}(?:{suffix})?')










arrayCableMarks = ['КУГПЭПнг', 
                    'SM-MLT', 
                    'КППГЭнг', 
                    'эп(А)-', 
                    'нг(А)', 
                    'КГН', 
                    'КРА6.560.106-27', 
                    'EC4-50-HF', 
                    'TППэп(A)-HДГ',
                    'СПЕЦЛАН',
                    'Complete',
                    'UNITRONIC',
                    'ТАДУ',
                    'KCнг(A)-HF',
                    '6XV1871-2F',
                    'Hyperline',
                    'СКАБ',
                    'PK75',
                    'PROFINET',
                    'F/UTP',
                    'FO-ST-IN/OUT',
                    'КАГЭнг(B)',
                    'нг(A)-HF',
                    'LLMD',
                    'ТК-1, тип Б исп.45',
                    'ТК-1, тип Б',
                    'Спецкабель',
                    'ТППэп',
                    'нг(A)',
                    'КПЭТИнг(B)-HF-Т',
                    'РКГМ',
                    'ППГнг(A)-HF',
                    'ОКБнг(В)-',
                    'OKБнг(B)-',
                    'ОБР-У-4ОВ SM',
                    'КУГПвЭПвнг(A)-HF-Т',
                    'КПЭТИнг(B)-FRHF-Т',
                    'RG58',
                    'RG213',
                    'KPEPng(А)-FRHF-0.25',
                    'KVVGEng(A)-FRLS',
                    'КПСВЭВнг(А)-LS',
                    'WDZA-KYJYP2-23',
                    'OKCнг(A)-FRHF',
                    'OKЗнг(A)-FRHF',
                    'ОКСнг(A)-FRHF',
                    'OKСнг(A)-FRHF',
                    'ОКЗнг(A)-FRHF',
                    'WDZA-KYJYP2-23'
                   ]


# ----- Заголовки таблицы Excel -----
HEADERS = [
    [0, 'Журнал', 40],
    [1, 'Номер кабеля', 9],
    [2, 'ККС', 21],
    [3, 'Группа', 7],
    [4, 'Марка', 22],
    [5, 'Сечение', 10],
    [6, 'Диаметр', 8],
    [7, 'Длина', 6],
    [8, 'Трасса', 20],
    [9, 'Откуда помещение', 14],
    [10, 'Откуда оборудование', 20],
    [11, 'Откуда x', 11],
    [12, 'Откуда y', 11],
    [13, 'Откуда z', 11],
    [14, 'Куда помещение', 14],
    [15, 'Куда оборудование', 20],
    [16, 'Куда x', 11],
    [17, 'Куда y', 11],
    [18, 'Куда z', 11],
    [19, 'Резервирование', 8],
    [20, 'Дата добавления в базу', 10],
    [21, 'Версия базы', 7],
    [22, 'id кабеля', 8],
    [23, 'Номер ревизии', 6],
    [24, 'Дата ревизии', 10],
    [25, 'Источник информации', 8],
    [26, 'Класс безопасности', 7],
    [27, 'Статус прокладки', 10],
    [28, 'Статус объединения', 10],
    [29, 'Raw', 25],
    [30, 'min длина', 10],
    [31, 'Требования к объединению', 20],
    [32, 'Ответная часть (из КЖ)', 40],
    [33, 'Наличие в базе', 20],    
    [34, 'Примечание', 40]
]